app/controllers/tests_controller.rb

```diff
@@ -0,0 +1,6 @@
+class TestsController < ApplicationController
+  def show
```

Comment:
It should be a custom `start` action. `show` is wrong because:
1. `show` is a method for retrieving a single resource.
2. Rails dispatches `show` via `GET` and get requests must be idempotent (i.e. they can't mutate your state)